### ISSUE TITLE
Implicit function compilation causes progressbar to 'freeze'

### ISSUE THREAD

**Issue by Pandoro** [lucasb-eyer/DeepFried2#11]
As discussed, the progressbar appears and seems to freeze at first. This could lead people to believe the program has died and it just looks weird.

**Comment by lucasb-eyer** [comment]
Yep, I want to add a function that can "precompile" a model/optimizer/criterion combo and use that in the examples, but I still need to think about the API.
